activedirectory/azure-ad-integration-spring-boot-autoconfigure-sample/README.md

```diff
@@ -0,0 +1,42 @@
+## Overview
+This sample illustrates how to use `azure-ad-integration-spring-boot-autoconfigure package` to plugin JWT token filter into Spring Security filter chain. The filter injects `UserPrincipal` object that is associated with the thread of the current user request. User's AAD membership info, along with token claimsset, JWS object etc. are accessible from the object which can be used for role based authorization. Methods like `isMemberOf` is also supported.
+
+### Get started
+The sample is composed of two layers: Angular JS client and Spring Boot RESTful Web Service. You need to make some changes to get it working with your Azure AD tenant on both sides.
+
+#### Application.properties
+You need to have an registered app in your Azure AD tenant and create a security key.
+Put Application ID and Key in `clientId` and `clientSecret` respectively e.g.
+`azure.activedirectory.clientId=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`
+`azure.activedirectory.clientSecret=ABCDEFGHIJKLMNOOPQRSTUVWXYZABCDEFGHIJKLMNOPQ`
+List all the AAD groups `aadGroups` that you want to have a Spring Security role object mapping to it. The role objects can then be used to manage access to resources that is behind Spring Security. e.g.
+`azure.activedirectory.aadGroups=group1,group2`
+You can use `@PreAuthorize` annotation or `UserPrincipal` to manage access to web API based on user's group membership. You will need to change `ROLE_group1` to groups you want to allow to access the API or you will get "Access is denied".
+
+##### Note: The sample retrieves user's group membership using Azure AD graph API which requires the registered app to have `Access the directory as the signed-in user` under `Delegated Permissions`. You need AAD admin privilege to be able to grant the permission in API ACCESS -> Required permission.
+
+
+#### Angular JS
+In `app.js`, make following changes. The client leverages Azure AD library for JS to handle AAD authentication in single page application. The following snippet of code configures adal provider for your registered app.
+```
+        adalProvider.init(
+            {
+                instance: 'https://login.microsoftonline.com/',
+                tenant: 'your-aad-tenant',
+                clientId: 'your-application-id',
+                extraQueryParameter: 'nux=1',
+                cacheLocation: 'localStorage',
+            },
+            $httpProvider
+        );
+
+```
+
+### Give it a run
+* Go to `path-to-azure-spring-boot-starters`, run `mvn clean package`.
+* `cd activedirectory\azure-ad-integration-spring-boot-autoconfigure`
+* `mvn install`
```

Comment:
if you do 'mvn package' in the root folder, i think it also runs without the mvn install